neqo-transport/src/path.rs

```diff
@@ -573,14 +578,37 @@ impl Path {
             challenge: None,
             rtt: RttEstimate::default(),
             sender,
-            tos: IpTos::default(), // TODO: Default to Ect0 when ECN is supported.
-            ttl: 64,               // This is the default TTL on many OSes.
+            tos: IpTosEcn::Ect0.into(),
+            ttl: 64, // This is the default TTL on many OSes.
             received_bytes: 0,
             sent_bytes: 0,
+            lost_ecn_packets: 0,
+            ecn_count: EnumMap::default(),
             qlog,
         }
     }
 
+    /// Return latest ECN count received on this path.
+    pub fn ecn_count(&self, space: PacketNumberSpace) -> EcnCount {
+        self.ecn_count[space]
+    }
+
+    /// Set the ECN count received on this path.
+    pub fn set_ecn_count(&mut self, space: PacketNumberSpace, ecn_count: EcnCount) {
+        qinfo!("Setting ECN count for {:} to {:?}", space, ecn_count);
+        self.ecn_count[space] = ecn_count;
+    }
+
+    /// Whether this path is currently marking packets with ECN.
+    pub fn is_ecn_enabled(&self) -> bool {
+        self.tos != IpTosEcn::NotEct.into()
```

Comment:
Is this right, or should this be:

```suggestion
        self.tos.ecn() == IpTosEcn::Ect0
```

Reply:
I think `IpTosEcn::from(self.tos) != IpTosEcn::NotEct` - we could just check if it's equal to `IpTosEcn::Ect0`, but maybe we'll mark `IpTosEcn::Ect1` at some point in the future.